examples/vision/image_captioning.py

```diff
@@ -384,60 +453,56 @@ def calculate_accuracy(self, y_true, y_pred, mask):
         mask = tf.cast(mask, dtype=tf.float32)
         return tf.reduce_sum(accuracy) / tf.reduce_sum(mask)
 
+    def _compute_caption_loss_and_acc(self, img_embed, batch_seq, training=True):
+        encoder_out = self.encoder(img_embed, training=training)
+        batch_seq_inp = batch_seq[:, :-1]
+        batch_seq_true = batch_seq[:, 1:]
+        mask = tf.math.not_equal(batch_seq_true, 0)
+        batch_seq_pred = self.decoder(
+            batch_seq_inp, encoder_out, training=training, mask=mask
+        )
+        loss = self.calculate_loss(batch_seq_true, batch_seq_pred, mask)
+        acc = self.calculate_accuracy(batch_seq_true, batch_seq_pred, mask)
+        return loss, acc
+
     def train_step(self, batch_data):
         batch_img, batch_seq = batch_data
         batch_loss = 0
         batch_acc = 0
 
         # 1. Get image embeddings
-        img_embed = self.cnn_model(batch_img)
+        img_embed = self.cnn_model(batch_img, training=False)
```

Comment:
Why `training=False`? Deserves a comment

Reply:
This was redundant actually. The base cnn model is already non-trainable, so we don't need this. I removed it